tool/tsh/tsh.go

```diff
@@ -1763,6 +1763,7 @@ func onJoin(cf *CLIConf) error {
 		return trace.Wrap(err)
 	}
 
+	cf.NodeLogin = teleport.SSHSessionJoinPrincipal
```

Comment:
Is this not going to affect regular `tsh join`?

Reply:
It is, that's the point of the PR. All joins use a special restricted principal and rely on moderated sessions for authz.